WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverViewModel.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+import UIKit
+import WordPressUI
+
+final class CompliancePopoverViewModel: ObservableObject {
+    @Published
+    var isAnalyticsEnabled: Bool = !WPAppAnalytics.userHasOptedOut()
+    var coordinator: CompliancePopoverCoordinator?
+
+    private let defaults: UserDefaults
+
+    init(defaults: UserDefaults) {
+        self.defaults = defaults
+    }
+
+    func didTapSettings() {
+        coordinator?.navigateToSettings()
+        defaults.shouldShowCompliancePopup = false
```

Comment:
Because the flag is toggled here and in `didTapSave()` only, when I dismiss the popover by tapping outside of it or by swiping it down, the popover is displayed the next time My Site is displayed.

My guess is this is not intentional. We should either disable dismissing the popover or avoid showing it again in all cases.

The second solution should be as simple as toggling the flag once the popover is displayed.

Reply:
I did that intentionally actually. It may not be aligned with Android but my logic was that people may want to get rid of the pop up and not read it if we don't allow them to skip it and force a reaction that they take without reading the message. But because they need to select something I show it until they respond. If that doesn't make sense we can change that.

Reply:
@alpavanoglu I understand your rationale now. However, I feel the popover is simple enough to understand and to take action with very quick skimming. Avoiding forcing an immediate response has its benefit, but as a user, I would expect that dismissing the popover is a sort of response (agreeing), or at least I'm delaying the response. So showing it immediately after can be quite annoying, in my opinion. In words, I see the cons outweighing the pros here.

We could add logic to show it after some time from the last dismissal, but that would be unnecessary extra scope. My recommendation is to disable other ways of dismissal, especially since the popover has a very clear CTA (the "Save" button) that dismisses it without any further action needed.

Reply:
As discussed, due to a problem in `BottomSheetViewController` (and class being public and not open) the solution to this is to implement `allowsDragToDismiss` in WordPressUI or make the class `open`. Doing the former will automatically make the pop-over not dismissible as the default value from protocol extension won't be used and the value will be received from `CompliancePopoverViewController`.